### PR TITLE
MINOR: Fix Streams-Broker-Compatibility system test

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py [hunk 1]
```diff
@@ -116,10 +114,9 @@ def test_timeout_on_pre_010_brokers(self, broker_version):
         self.kafka.start()
 
         processor = StreamsBrokerCompatibilityService(self.test_context, self.kafka, False)
-        processor.start()
 
-        processor.node.account.ssh(processor.start_cmd(processor.node))
         with processor.node.account.monitor_log(processor.STDERR_FILE) as monitor:
+            processor.start()
```
Comment (guozhangwang) [inline tests/kafkatest/tests/streams/streams_broker_compatibility_test.py:119]: Is this change intentional?

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py [hunk 2]
```diff
@@ -98,13 +97,12 @@ def test_fail_fast_on_incompatible_brokers(self, broker_version):
         self.kafka.start()
 
         processor = StreamsBrokerCompatibilityService(self.test_context, self.kafka, False)
-        processor.start()
 
-        processor.node.account.ssh(processor.start_cmd(processor.node))
         with processor.node.account.monitor_log(processor.STDERR_FILE) as monitor:
-            monitor.wait_until('FATAL: An unexpected exception org.apache.kafka.streams.errors.StreamsException: Could not create topic kafka-streams-system-test-broker-compatibility-KSTREAM-AGGREGATE-STATE-STORE-0000000001-changelog.',
+            processor.start()
```
Comment (guozhangwang) [inline tests/kafkatest/tests/streams/streams_broker_compatibility_test.py:102]: Ditto here.

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py [hunk 3]
```diff
@@ -63,13 +63,12 @@ def test_fail_fast_on_incompatible_brokers_if_eos_enabled(self, broker_version):
         self.kafka.start()
 
         processor = StreamsBrokerCompatibilityService(self.test_context, self.kafka, True)
-        processor.start()
 
-        processor.node.account.ssh(processor.start_cmd(processor.node))
         with processor.node.account.monitor_log(processor.STDERR_FILE) as monitor:
-            monitor.wait_until("Exception in thread \"main\" org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support LIST_OFFSETS ",
+            processor.start()
```
Comment (guozhangwang) [inline tests/kafkatest/tests/streams/streams_broker_compatibility_test.py:68]: Ditto here.

File: streams/src/test/java/org/apache/kafka/streams/tests/BrokerCompatibilityTest.java [hunk 4]
```diff
@@ -133,15 +144,15 @@ private static void loopUntilRecordReceived(final String kafka, final boolean eo
             consumerProperties.put(ConsumerConfig.ISOLATION_LEVEL_CONFIG, IsolationLevel.READ_COMMITTED.name().toLowerCase(Locale.ROOT));
         }
 
-        final KafkaConsumer<String, String> consumer = new KafkaConsumer<>(consumerProperties);
-        consumer.subscribe(Collections.singletonList(SINK_TOPIC));
+        try(final KafkaConsumer<String, String> consumer = new KafkaConsumer<>(consumerProperties)) {
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/tests/BrokerCompatibilityTest.java]: nit: space after `try`.

**Review by guozhangwang — COMMENTED** [review]
Just synced with hachikuji offline, I think https://issues.apache.org/jira/browse/KAFKA-6289 may not be the root cause of this, but probably KAFKA-6397.

**Comment by mjsax** [comment]
> Just synced with hachikuji offline, I think https://issues.apache.org/jira/browse/KAFKA-6289 may not be the root cause of this, but probably KAFKA-6397.

Maybe. However, it seems that KAFKA-6397 might have been the root cause for the failing EOS tests (I assume that the first PR did not contain the change and thus passed -- and hence, failed again after merging -- the second PR change `poll()` to a for-loop calling `position()` instead...

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py (hunk 3 above)
Comment (mjsax) [reply tests/kafkatest/tests/streams/streams_broker_compatibility_test.py:68]: Yes. This is a race condition -- we always need to get the monitor first -- also unclear to me, why we start the processor twice in the old code...

**Comment by mjsax** [comment]
System test passed: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/1319/

Triggered one more build: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/1333/

**Comment by mjsax** [comment]
Retest this please.

**Comment by mjsax** [comment]
Retest this please.

**Comment by mjsax** [comment]
\cc @bbejeck

**Comment by mjsax** [comment]
Retest this please.

**Review by bbejeck — APPROVED** [review]
LGTM

**Review by guozhangwang — APPROVED** [review]

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `1.1` branch.
